cpp/open3d/geometry/TriangleMesh.cpp

```diff
@@ -478,6 +479,18 @@ std::shared_ptr<PointCloud> TriangleMesh::SamplePointsUniformlyImpl(
                                       b * vertex_colors_[triangle(1)] +
                                       c * vertex_colors_[triangle(2)];
         }
+        // if (has_triangle_uvs && has_textures) {
+            Eigen::Vector2d uv = a * triangle_uvs_[triangle(0)] + 
+                    b * triangle_uvs_[triangle(1)] + 
+                    c * triangle_uvs_[triangle(2)];
+
+            int w = textures_[0].width_;
+            int h = textures_[0].height_;
+            pcd->colors_[point_idx] = Eigen::Vector3d(
+                        (double) *textures_[0].PointerAt<uint8_t>(uv(0) * w, uv(1) * h, 0) / 255 , 
+                        (double) *textures_[0].PointerAt<uint8_t>(uv(0) * w, uv(1) * h, 1) / 255 , 
+                        (double) *textures_[0].PointerAt<uint8_t>(uv(0) * w, uv(1) * h, 2) / 255);
```

Comment:
@benjaminum 
I'm not an Computer graphics expert. It seems the barycentric coordinates you were referring to are a sort of mixture weights for triangles (in this case it looks like a,b,c ?). Unfortunately this code seems to samples from the correct colors but from the wrong position in the texture map. 

Am I making a mistake here with the `uv` computation or am I using the `PointerAt` function wrong ?
Maybe you see a mistake immediately?

Reply:
Can you try `(1-uv(1))*h`?

Reply:
also does not look quite right unfortunately:
![test_1_minus](https://github.com/isl-org/Open3D/assets/11996097/17eaa3d8-bfcb-4a3b-9a54-462dd907ffb3)

Reply:
could it be that `triangle(0), triangle(1), triangle(2)` in `triangle_uvs_`don't select adjacent triangle vertices in the texture? 
I'm trying to run it on a cube with different surface colors. This is the texture map:
![texture_triangle_mapping](https://github.com/isl-org/Open3D/assets/11996097/24d761ea-ff29-4851-be55-75a1a7b8434f)

Reply:
I suppose the `triangle_index_generator` does not take this into account, or something is unexpected in `triangle_uvs_`

Reply:
@nikste triangle stores the indices of the vertices but we need the index to the right uv.
`triangle_uvs_` if I recall correctly should have the size 3*#triangles. Can you try to get the uvs with [3*tidx, 3*tidx+1, 3*tidx+2]?

Reply:
@benjaminum  
Great! It works now. Thanks for your help!

Couple of questions: 
1. Why is `textures_` a vector, does it always has only one element ? Do we need to add some different behavior here for multiple elements in `textures_`?
2. Is there a case that the object could have `HasVertexColors()` and `HasTextures()` both are true?